src/base/Assert.hh

```diff
@@ -33,6 +35,13 @@
  * Postcondition debug assertion macro. Use to "ensure" that return values or
  * side effects are as expected when leaving a function.
  */
+/*!
+ * \def INSIST
+ *
+ * Always-on runtime assertion macro. This can check user input and input data
+ * consistency, and will raise RuntimeError on failure with a descriptive error
+ * message. This should not be used on device.
```

Comment:
Why "this" macro should not be used on device (why this one and not the other)?  Or do we mean "avoid in time critical/sensitive code" ?

Reply:
The real reason is that the `INSIST` macro allows streaming operators (using the host-only `ostringstream`) to allow verbose output. But you're right that it's intended for setup code and should be avoided in time-critical code.